src/ppx/grfx/vk/vk_descriptor.cpp

```diff
@@ -275,11 +276,20 @@ Result DescriptorSetLayout::CreateApiObjects(const grfx::DescriptorSetLayoutCrea
         vkBinding.stageFlags                   = ToVkShaderStageFlags(baseBinding.shaderVisiblity);
         vkBinding.pImmutableSamplers           = nullptr;
         vkBindings.push_back(vkBinding);
+
+        VkDescriptorBindingFlags vkBindingFlag = ToVkDescriptorBindingFlags(baseBinding.flags);
+        vkBindingFlags.push_back(vkBindingFlag);
     }
 
+    VkDescriptorSetLayoutBindingFlagsCreateInfoEXT vkBindingWrapper = {VK_STRUCTURE_TYPE_DESCRIPTOR_SET_LAYOUT_BINDING_FLAGS_CREATE_INFO_EXT};
+    vkBindingWrapper.bindingCount                                   = CountU32(vkBindingFlags);
+    vkBindingWrapper.pBindingFlags                                  = DataPtr(vkBindingFlags);
+    vkBindingWrapper.pNext                                          = nullptr;
```

Comment:
Do we add this even if none of the flags are set? 
It seems like it comes from VK_EXT_descriptor_indexing extension, which is optional currently https://github.com/google/bigwheels/blob/a99d5bfbd4127f41677fe7113447aae587e90d4f/src/ppx/grfx/vk/vk_device.cpp#L154 so it's possible this extension won't be enabled. 
Perhaps we should only set this if the flags are actually not 0? And assert that the required features are supported?

Reply:
Right - my mistake, I did mean to call the function and check for support. Will add this